# Gaming marketing dashboard

### Dataset: video games sales data 2024 (vg_data_24.csv renamed) from kaggle
### purpose: exploring the dataset to see what data we have and cleaning where appropirate.


## 1. Setup - Importing libraries

In [1]:
# Importing libraries

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import os

# Setting display options for pandas
pd.set_option('display.max_columns', None)

# Setting up plotting style
plt.style.use('seaborn-v0_8-darkgrid')

print ('libraries loaded and set up complete')

libraries loaded and set up complete


## 2. Loading dataset 

In [6]:
# reading data from data /raw folder
df = pd.read_csv('../data/raw/vg_data_24.csv')

# quick look at the data
print(f'rows: {df.shape[0]}, columns: {df.shape[1]}')

rows: 64016, columns: 14


## 3.1 Exploring data - looking at intial data

looking at what we have on the data

In [22]:
# first five rows of the data
df.head()



img                        title  \
0  /games/boxart/full_6510540AmericaFrontccc.jpg           Grand Theft Auto V   
1  /games/boxart/full_5563178AmericaFrontccc.jpg           Grand Theft Auto V   
2                    /games/boxart/827563ccc.jpg  Grand Theft Auto: Vice City   
3  /games/boxart/full_9218923AmericaFrontccc.jpg           Grand Theft Auto V   
4  /games/boxart/full_4990510AmericaFrontccc.jpg    Call of Duty: Black Ops 3   

  console    genre       publisher       developer  critic_score  total_sales  \
0     PS3   Action  Rockstar Games  Rockstar North           9.4        20.32   
1     PS4   Action  Rockstar Games  Rockstar North           9.7        19.39   
2     PS2   Action  Rockstar Games  Rockstar North           9.6        16.15   
3    X360   Action  Rockstar Games  Rockstar North           NaN        15.86   
4     PS4  Shooter      Activision        Treyarch           8.1        15.09   

   na_sales  jp_sales  pal_sales  other_sales release_date last_update  
0      6.37      0.99       9.85         3.12   2013-09-17         NaN  
1      6.06      0.60       9.71         3.02   2014-11-18  2018-01-03  
2      8.41      0.47       5.49         1.78   2002-10-28         NaN  
3      9.06      0.06       5.33         1.42   2013-09-17         NaN  
4      6.18      0.41       6.05         2.44   2015-11-06  2018-01-14

In [23]:
# checking what data we have overall
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 64016 entries, 0 to 64015
Data columns (total 14 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   img           64016 non-null  object 
 1   title         64016 non-null  object 
 2   console       64016 non-null  object 
 3   genre         64016 non-null  object 
 4   publisher     64016 non-null  object 
 5   developer     63999 non-null  object 
 6   critic_score  6678 non-null   float64
 7   total_sales   18922 non-null  float64
 8   na_sales      12637 non-null  float64
 9   jp_sales      6726 non-null   float64
 10  pal_sales     12824 non-null  float64
 11  other_sales   15128 non-null  float64
 12  release_date  56965 non-null  object 
 13  last_update   17879 non-null  object 
dtypes: float64(6), object(8)
memory usage: 6.8+ MB


In [24]:
# column names
print("Column names:")
df.columns.tolist()

Column names:


['img',
 'title',
 'console',
 'genre',
 'publisher',
 'developer',
 'critic_score',
 'total_sales',
 'na_sales',
 'jp_sales',
 'pal_sales',
 'other_sales',
 'release_date',
 'last_update']

### Observation so far
- consists of 64016 rows 
- consists of 14 columns in total
- data type consists of objects and floats
- has sales data for different regions
- critic score and other key columns like genre, publishers, developers and consoles is present

### 3.2 Missing Data Check

Need to see where we have gaps in the data.

In [39]:

# cheking for missing values
missing = pd.DataFrame({
    'Columns': df.columns,
    'Total': len(df),
    'Present': df.count(),
    'Missing': df.isnull().sum()
})
# sortung by most missing values
missing = missing.sort_values('Missing', ascending=False)

# displaying without index
print(missing.to_string(index=False))




     Columns  Total  Present  Missing
critic_score  64016     6678    57338
    jp_sales  64016     6726    57290
    na_sales  64016    12637    51379
   pal_sales  64016    12824    51192
 other_sales  64016    15128    48888
 last_update  64016    17879    46137
 total_sales  64016    18922    45094
release_date  64016    56965     7051
   developer  64016    63999       17
   publisher  64016    64016        0
         img  64016    64016        0
       genre  64016    64016        0
     console  64016    64016        0
       title  64016    64016        0


### Observation so far
- 9 columns with missing values that needs attention
- img column is not needed will remove 
- sales data missing for many games across all genre, which makes sense as not all games get reviews


### 3.3 Exploring key columns


### Platforms

In [ ]:
# deifferent platforms we have in the data
print(f"Number of platforms: {df['console'].nunique()}")

# Top 15 platforms
print("\nTop 15 platforms: \n")
print(df['console'].value_counts().head(15))

Number of platforms: 81

Top 15 platforms: 

console
PC      12617
PS2      3565
DS       3288
PS4      2878
PS       2707
NS       2337
XBL      2120
PSN      2004
XOne     1963
PS3      1905
PSP      1807
X360     1735
Wii      1676
GBA      1659
GB       1599
Name: count, dtype: int64


### Genre

In [ ]:
# different genres arewe have in the data
print(f"Number of genres: {df['genre'].nunique()}\n")

# Top genres
print(df['genre'].value_counts())

Number of genres: 20

genre
Misc                9304
Action              8557
Adventure           6260
Role-Playing        5721
Sports              5586
Shooter             5410
Platform            4001
Strategy            3685
Puzzle              3521
Racing              3425
Simulation          3158
Fighting            2367
Action-Adventure    1877
Visual Novel         493
Music                297
Party                151
MMO                  115
Education             35
Board Game            33
Sandbox               20
Name: count, dtype: int64


### 3.4 Exploring Sales Data

Looking at the sales figures for games that have in this data.

In [ ]:
# Stats for all sales columns (unit in millions)
sales_columns = ['total_sales', 'na_sales', 'jp_sales', 'pal_sales', 'other_sales']
print(df[sales_columns].describe())

        total_sales      na_sales     jp_sales     pal_sales   other_sales
count  18922.000000  12637.000000  6726.000000  12824.000000  15128.000000
mean       0.349113      0.264740     0.102281      0.149472      0.043041
std        0.807462      0.494787     0.168811      0.392653      0.126643
min        0.000000      0.000000     0.000000      0.000000      0.000000
25%        0.030000      0.050000     0.020000      0.010000      0.000000
50%        0.120000      0.120000     0.040000      0.040000      0.010000
75%        0.340000      0.280000     0.120000      0.140000      0.030000
max       20.320000      9.760000     2.130000      9.850000      3.120000


### Observation so far
- Average game sells about 0.35 million copies (approx 350k copies sold)
- Top games reach total of 20+ million
- Most games sell way less than the average (median is 0.12m which is about 120k copies)

## 4. Cleaning the Data

Now I'll clean up the data to make it ready for analysis.

### 4.1 Remove columns we don't need

In [ ]:
# Drop image path and last_update - not useful for analysis
df = df.drop(columns=['img', 'last_update'])
print(f"Columns remaining: {df.shape[1]}")


Columns remaining: 12


### 4.2 Fixing release dates and extracting year

In [62]:
#convert release_date to datetime
df['release_date'] = pd.to_datetime(df['release_date'], errors='coerce')

In [ ]:
# creating a column to extract year from release_date
df['release_year'] = df['release_date'].dt.year

# checking range of years in the data (removing float values)
print(f"Years range from {df['release_year'].min():.0f} to {df['release_year'].max():.0f}")

Years range from 1971 to 2024


### 4.3 Key focus on recent data between (2010-2024)

For the dashboard, I'll focus on the last 14 years to show modern gaming trends.

In [ ]:
# only keeping games from 2010 onwards
df_clean = df[df['release_year'] >= 2010].copy()

print(f"Original dataset: {len(df):,} games")
print(f"After filtering: {len(df_clean):,} games")
print(f"Removed: {len(df) - len(df_clean):,} older games")

Original dataset: 64,016 games
After filtering: 22,782 games
Removed: 41,234 older games


## 5. Saving cleaned dataset


In [79]:

# saving data to clean folder
df_clean.to_csv('..\\data\\clean\\vg_data_cleaned.csv', index=False)
print("cleaned data saved")
print(f"final dataset shape: {df_clean.shape} (rows, columns)")

cleaned data saved
final dataset shape: (22782, 13) (rows, columns)


### Summary of changes made:

1. Loaded 64,016 games from the raw dataset
2. Explored the data structure and found key columns
3. Identified missing data (expected for reviews)
4. Removed irrelavent columns (image, update)
5. Converted release dates and extracted in years
6. Focused on data from 2010-2024 for modern gaming analysis
7. Saved cleaned dataset for dashboard